server/src/main/java/org/opensearch/index/mapper/MatchOnlyTextFieldMapper.java

```diff
@@ -0,0 +1,310 @@
+            if (fieldData.getValue()) {
+                ft.setFielddata(true, freqFilter.getValue());
+            }
+            return ft;
+        }
+
+        @Override
+        protected List<Parameter<?>> getParameters() {
+            return Arrays.asList(
+                index,
+                store,
+                indexOptions,
+                norms,
+                termVectors,
+                analyzers.indexAnalyzer,
+                analyzers.searchAnalyzer,
+                analyzers.searchQuoteAnalyzer,
+                similarity,
+                positionIncrementGap,
+                fieldData,
+                freqFilter,
+                eagerGlobalOrdinals,
+                indexPhrases,
+                indexPrefixes,
+                boost,
+                meta
+            );
+        }
+    }
+
+    /**
+     * The specific field type for MatchOnlyTextFieldMapper
+     *
+     * @opensearch.internal
+     */
+    public static final class MatchOnlyTextFieldType extends TextFieldMapper.TextFieldType {
+        private final boolean indexPhrases = false;
+
+        private PrefixFieldType prefixFieldType;
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        public MatchOnlyTextFieldType(String name, boolean indexed, boolean stored, TextSearchInfo tsi, Map<String, String> meta) {
+            super(name, indexed, stored, tsi, meta);
+        }
+
+        @Override
+        public Query phraseQuery(TokenStream stream, int slop, boolean enablePosIncrements, QueryShardContext context) throws IOException {
+            PhraseQuery phraseQuery = (PhraseQuery) super.phraseQuery(stream, slop, enablePosIncrements);
+            BooleanQuery.Builder builder = new BooleanQuery.Builder();
+            for (Term term : phraseQuery.getTerms()) {
+                builder.add(new TermQuery(term), BooleanClause.Occur.FILTER);
+            }
+            return new SourceFieldMatchQuery(builder.build(), phraseQuery, this, context);
+        }
+
+        @Override
+        public Query multiPhraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements, QueryShardContext context)
+            throws IOException {
+            MultiPhraseQuery multiPhraseQuery = (MultiPhraseQuery) super.multiPhraseQuery(stream, slop, enablePositionIncrements);
+            BooleanQuery.Builder builder = new BooleanQuery.Builder();
+            for (Term[] terms : multiPhraseQuery.getTermArrays()) {
+                if (terms.length > 1) {
+                    BooleanQuery.Builder disjunctions = new BooleanQuery.Builder();
+                    for (Term term : terms) {
+                        disjunctions.add(new TermQuery(term), BooleanClause.Occur.SHOULD);
+                    }
+                    builder.add(disjunctions.build(), BooleanClause.Occur.FILTER);
+                } else {
+                    builder.add(new TermQuery(terms[0]), BooleanClause.Occur.FILTER);
```

Comment:
Would be nice to have some comments to explain why we need disjunctions and the boolean clauses for posterity.

Reply:
done